tensorflow/python/data/kernel_tests/cache_test.py

```diff
@@ -170,6 +171,19 @@ def testReadingPastEndOfSequence(self):
     expected_output = [[0, 1, 2, 3], [4, 5, 6, 7], [8, 9]] * 2
     self.assertDatasetProduces(dataset, expected_output)
 
+  def testCleaningUpCacheFiles(self):
+    def do_test(i):
+      dataset = dataset_ops.Dataset.range(10).cache(self.cache_prefix)
+      get_next = self.getNext(dataset)
+      for _ in range(i):
+        try:
+          self.evaluate(get_next())
+        except errors.OutOfRangeError:
+          break
+
+    if context.executing_eagerly():
```

Comment:
does this not work in graph mode?

Reply:
This test case does not work in graph mode. In my understanding, it is because the iterator resource is still held when the iteration is not finished, so the iterator deconstructor can not be triggered. The error message is in below. This failure may be expected.
```
There appears to be a concurrent caching iterator running - cache lockfile already exists ('/var/folders/44/dzfy03t93h990mbh3qjy3pjm0000gn/T/tmp6bpwffxz/cache_0.lockfile'). If you are sure no other running TF computations are using this cache prefix, delete the lockfile and re-initialize the iterator. Lockfile contents: Created at: 1563470964
```

Reply:
Got it, makes sense. Can we update the check to
```python
if not context.executing_eagerly():
  self.skipTest("Test requires eager mode for iterators to be deconstructed")

for i in [0, 3, 10, 12, 15]:
  do_test(i)
```